minkindr/test/test-transformation-2d.cc

```diff
@@ -0,0 +1,123 @@
+#include <cmath>
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <gtest/gtest.h>
+
+#include "kindr/minimal/transform-2d.h"
+
+#ifndef TEST
+#define TEST(a, b) int Test_##a##_##b()
+#endif
+
+#define __INTERNAL_GTEST_NEAR_EIGEN(PREDICATE, matrix_A, matrix_B, precision) \
+  PREDICATE##_TRUE(                                                           \
+      ((matrix_A) - (matrix_B)).cwiseAbs().maxCoeff() <= precision)           \
+      << "For matrices '" << #matrix_A << "' and '" << #matrix_B << "'."      \
+      << std::endl                                                            \
+      << "Where '" << #matrix_A << "' equals: " << std::endl                  \
+      << (matrix_A) << std::endl                                              \
+      << "and '" << #matrix_B << "' equals: " << std::endl                    \
+      << (matrix_B) << std::endl                                              \
+      << "and precision equals: " << precision;
+
+#define EXPECT_NEAR_EIGEN(matrix_A, matrix_B, precision) \
+  __INTERNAL_GTEST_NEAR_EIGEN(EXPECT, matrix_A, matrix_B, precision)
+
+namespace kindr {
+namespace minimal {
+
+constexpr double kEpsilon = 1e-10;
+static const Rotation2D r(0.77);
```

Comment:
I would still change this to capital `R` as this is used as a matrix; it doesn't matter how it's implemented in the back.

Reply:
but I think it's nicer to have this consistent with the member name that the rotation component of the `Transformation2D` class has

Reply:
Then maybe that should be also uppercase `R`? What are our naming guidelines with regards to matrices? We use `T_A_B` and so on, but I don't know about other cases.

Reply:
I would prefer to call it lower-case `r`:

```
Transformation2DTemplate(const Rotation r_A_B, const Position& A_t_A_B);
```

With `R`, one would rather assume that this is a 2x2 Eigen Matrix (which is not the case):

```
Transformation2DTemplate(const Rotation R_A_B, const Position& A_t_A_B);
```

I see that a small `r` is definitely not the most unique member name, but also I would say that it's pretty obvious what it is given the context, and also it's similar to the naming here:

https://github.com/ethz-asl/minkindr/blob/master/minkindr/include/kindr/minimal/quat-transformation.h#L185-L188

Having said that, I would also be fine changing it to `R`, I think it's rather a matter of personal taste. Any opinions @HannesSommer ?

Reply:
For this local variable of a test implementation I think we should not spend too much time on it and keep it as is -- just because it is the least effort. For a member things are different but not that much since it is a private member. 

For the general question of using capital letters for matrices and whether this is an instance of a matrix in that sense the mathematician in me can only say: this is exactly the sort of problem we get with assigning capital letters randomly to some unimportant subclass ob mathematical objects :) (sorry, you are of course not guilty of this convention ..)
Having said that, I believe this not to be a matrix in this sense and would therefore go for a lower case `r`, just like the quaternions is usually written with a lower case letter even if it is used as representative of a rotation.

Reply:
Thanks for the clarifications @HannesSommer !